updatehub/tests/integration_test.rs

```diff
@@ -0,0 +1,427 @@
+        .assert();
+
+    println!("{:#?}", assert.get_output());
+}
+
+#[test]
+fn correct_config_with_polling_enabled_no_update() {
+    use assert_cmd::Command;
+    use std::{fs::File, io::Write};
+    use tempfile::tempdir;
+
+    let path = updatehub::firmware::tests::create_fake_metadata(
+        updatehub::firmware::tests::FakeDevice::NoUpdate,
+    );
+    let dir = tempdir().unwrap();
+    let file_path = dir.path().join("updatehub.conf");
+    let temp_dir = file_path.clone();
+    let mut file = File::create(file_path).unwrap();
+    write!(file, "{}", write_basic_config(path)).unwrap();
+
+    let mut cmd = Command::cargo_bin("updatehub").unwrap();
+    let assert = cmd
+        .arg("server")
+        .arg("-v")
+        .arg("trace")
+        .arg("-c")
+        .arg(temp_dir)
+        .timeout(std::time::Duration::from_secs(1))
+        .env("sleep", "100")
+        .assert();
+
+    println!("{:#?}", assert.get_output());
+}
+
+#[test]
+fn correct_config_with_polling_disabled_no_update() {
+    use assert_cmd::Command;
+    use std::{fs::File, io::Write};
+    use tempfile::tempdir;
+
+    let path = updatehub::firmware::tests::create_fake_metadata(
+        updatehub::firmware::tests::FakeDevice::NoUpdate,
+    );
+    let dir = tempdir().unwrap();
+    let file_path = dir.path().join("updatehub.conf");
+    let temp_dir = file_path.clone();
+    let mut file = File::create(file_path).unwrap();
+    write!(
+        file,
+        "{}",
+        format!(
+            r#"
+[network]
+server_address="https://api.updatehub.io"
+listen_socket="localhost:8000"
+
+[storage]
+read_only=false
+runtime_settings="/data/updatehub/state.data"
+
+[polling]
+enabled=false
+interval="60s"
+
+[update]
+download_dir="/tmp/updatehub"
+supported_install_modes=["copy", "tarball"]
+
+[firmware]
+metadata="{}""#,
+            path.to_string_lossy()
+        )
+    )
+    .unwrap();
+
+    let mut cmd = Command::cargo_bin("updatehub").unwrap();
+    let assert = cmd
+        .arg("server")
+        .arg("-v")
+        .arg("trace")
+        .arg("-c")
+        .arg(temp_dir)
+        .timeout(std::time::Duration::from_secs(1))
+        .env("sleep", "100")
+        .assert();
+
+    println!("{:#?}", assert.get_output());
+}
+
+#[test]
+fn correct_config_with_polling_probe_no_update() {
+    use assert_cmd::Command;
+    use std::{fs::File, io::Write};
+    use tempfile::tempdir;
+
+    let path = updatehub::firmware::tests::create_fake_metadata(
+        updatehub::firmware::tests::FakeDevice::NoUpdate,
+    );
+    let dir = tempdir().unwrap();
+    let file_path = dir.path().join("updatehub.conf");
+    let mut file = File::create(file_path).unwrap();
+    write!(file, "{}", write_basic_config(path)).unwrap();
+
+    let mut cmd = Command::cargo_bin("updatehub").unwrap();
+    let assert = cmd
+        .arg("client")
+        .arg("probe")
+        .timeout(std::time::Duration::from_secs(1))
+        .env("sleep", "100")
+        .assert();
```

Comment:
Is the updatehub agent running? The probe is sent to a running instance of agent. No point in probing if it's not running.